docarray/base_document/mixins/proto.py

```diff
@@ -15,75 +14,147 @@
+                a.to_protobuf(), field_map={'url': 'link', 'tensor': 'array'}
             )
 
-            if content_type in content_type_dict:
-                fields[field] = content_type_dict[content_type].from_protobuf(
-                    getattr(value, content_key)
-                )
-            elif content_key == 'document':
-                fields[field] = cls._get_field_type(field).from_protobuf(
-                    value.document
-                )  # we get to the parent class
-            elif content_key == 'document_array':
-                from docarray import DocumentArray
-
-                fields[field] = DocumentArray.from_protobuf(
-                    value.document_array
-                )  # we get to the parent class
-            elif content_key is None:
-                fields[field] = None
-            elif content_type is None:
-                if content_key == 'text':
-                    fields[field] = value.text
-                elif content_key == 'blob':
-                    fields[field] = value.blob
-                elif content_key == 'integer':
-                    fields[field] = value.integer
-                elif content_key == 'float':
-                    fields[field] = value.float
-                elif content_key == 'boolean':
-                    fields[field] = value.boolean
-                elif content_key == 'list':
-                    from google.protobuf.json_format import MessageToDict
-
-                    fields[field] = MessageToDict(value.list)
-                elif content_key == 'set':
-                    from google.protobuf.json_format import MessageToDict
-
-                    fields[field] = set(MessageToDict(value.set))
-                elif content_key == 'tuple':
-                    from google.protobuf.json_format import MessageToDict
-
-                    fields[field] = tuple(MessageToDict(value.tuple))
-                elif content_key == 'dict':
-                    from google.protobuf.json_format import MessageToDict
-
-                    fields[field] = MessageToDict(value.dict)
+            assert b.link == a.url
+            assert (b.array == a.tensor).all()
+
+        :param pb_msg: the proto message of the Document
+        :param field_map: map proto key to schema key
+        :return: a Document initialize with the proto data
+        """
+
+        fields: Dict[str, Any] = {}
+
+        field_map = field_map or {}
+
+        for field_name in pb_msg.data:
+
+            do_pass_loop = False
+            if field_name not in cls.__fields__.keys():
+                if field_name in field_map.keys():
+                    field_name_mapped = field_map[field_name]
                 else:
-                    raise ValueError(
-                        f'key {content_key} is not supported for' f' deserialization'
-                    )
+                    do_pass_loop = True  # optimization we don't even load the data if the key does not
```

Comment:
I think we can remove the entire `do_pass_loop` variable and logic and instead `continue` here

Reply:
shit I know I was doing something wrong I was using pass instead of continue